xbmc/pvr/PVRThumbLoader.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ *  Copyright (C) 2005-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "PVRThumbLoader.h"
+
+#include "FileItem.h"
+#include "ServiceBroker.h"
+#include "TextureCache.h"
+#include "TextureCacheJob.h"
+#include "pictures/Picture.h"
+#include "pvr/PVRGUIDirectory.h"
+#include "pvr/PVRManager.h"
+#include "settings/AdvancedSettings.h"
+#include "settings/Settings.h"
+#include "settings/SettingsComponent.h"
+#include "utils/StringUtils.h"
+#include "utils/log.h"
+
+#include <ctime>
+
+using namespace PVR;
+
+bool CPVRThumbLoader::LoadItem(CFileItem* item)
+{
+  bool result = LoadItemCached(item);
+  result |= LoadItemLookup(item);
+  return result;
+}
+
+bool CPVRThumbLoader::LoadItemCached(CFileItem* item)
+{
+  return FillThumb(*item);
+}
+
+bool CPVRThumbLoader::LoadItemLookup(CFileItem* item)
+{
+  return false;
+}
+
+void CPVRThumbLoader::OnLoaderFinish()
+{
+  if (m_bInvalidated)
+  {
+    m_bInvalidated = false;
+    CServiceBroker::GetPVRManager().PublishEvent(PVREvent::ChannelGroupsInvalidated);
+  }
+  CThumbLoader::OnLoaderFinish();
+}
+
+void CPVRThumbLoader::ClearCachedImage(CFileItem& item)
+{
+  const std::string thumb = item.GetArt("thumb");
+  if (!thumb.empty())
+  {
+    CTextureCache::GetInstance().ClearCachedImage(thumb);
+    if (m_textureDatabase->Open())
+    {
+      m_textureDatabase->ClearTextureForPath(item.GetPath(), "thumb");
+      m_textureDatabase->Close();
+    }
+    item.SetArt("thumb", "");
+    m_bInvalidated = true;
+  }
+}
+
+void CPVRThumbLoader::ClearCachedImages(CFileItemList& items)
+{
+  for (auto& item : items)
+    ClearCachedImage(*item);
+}
+
+bool CPVRThumbLoader::FillThumb(CFileItem& item)
```

Comment:
This method always returns true, maybe use void instead?

Reply:
Oh, it's meant to return false if no thumb was found/loaded/created. Fixed now.